pkg/controllers/workgenerator/controller.go

```diff
@@ -850,6 +884,9 @@ func buildAllWorkAppliedCondition(works map[string]*fleetv1beta1.Work, binding *
 			ObservedGeneration: binding.GetGeneration(),
 		}
 	}
+	// In the case where ReportDiff apply strategy is used, work applier will no longer update
+	// Applied condition on Work objects; since existing Applied conditions will all become stale,
+	// this function will return a False Applied condition.
```

Comment:
need an e2e test to verify and safe guard this behavior.

Reply:
Yeah, will add one in the later PR.